src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.next = next;
+            this.value = value;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNode(index);
+        Node<T> prevNode = node.prev;
+        Node<T> newNode = new Node<>(prevNode, value, node);
+        node.prev = newNode;
+        if (prevNode == null) {
+            first = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list.size() == 0) {
+            return false;
+        }
+        for (T node : list) {
+            add(node);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> changedNode = getNode(index);
+        T oldValue = changedNode.value;
+        changedNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> removedNode = getNode(index);
+        T value = removedNode.value;
+        unlink(removedNode);
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = first;
+        for (int i = 0; i < size; i++) {
+            if (node.value == null || node.value.equals(object)) {
```

Comment:
Does it work well? How about 
```suggestion
            if (object == node.value || object != null && object.equals(node.value)) {
```